tests/test_network.py

```diff
@@ -124,6 +134,159 @@ def test_add_objective(factory, client):
+    spec = method_create(metadata={"foo": too_long})
+    with pytest.raises(substra.exceptions.InvalidRequest):
+        method_add(spec)
+
+
+@pytest.mark.parametrize('asset_name, algo_type', [
+    ('traintuple', 'algo'),
+    ('composite_traintuple', 'composite_algo'),
+])
+def test_metadata_traintuple(factory, client, asset_name, algo_type, default_dataset):
+    method_create = getattr(factory, f"create_{asset_name}")
+    method_add = getattr(client, f"add_{asset_name}")
+
+    algo_create = getattr(factory, f"create_{algo_type}")
+    algo_add = getattr(client, f"add_{algo_type}")
+
+    algo_spec = algo_create()
+    algo = algo_add(algo_spec)
+    spec = method_create(
+        algo=algo,
+        dataset=default_dataset,
+        data_samples=default_dataset.train_data_sample_keys,
+        metadata={"foo": "bar"}
+    )
+    asset = method_add(spec)
+
+    assert asset.metadata == {"foo": "bar"}
+
+    algo_spec = algo_create()
+    algo = algo_add(algo_spec)
+    spec = method_create(
+        algo=algo,
+        dataset=default_dataset,
+        data_samples=default_dataset.train_data_sample_keys,
+        metadata={}
+    )
+    asset = method_add(spec)
+
+    assert asset.metadata == {}
+
+    algo_spec = algo_create()
+    algo = algo_add(algo_spec)
+    spec = method_create(
+        algo=algo,
+        dataset=default_dataset,
+        data_samples=default_dataset.train_data_sample_keys,
+        metadata=None
+    )
+    asset = method_add(spec)
+
+    assert asset.metadata == {}
+
+    spec = method_create(
+        algo=algo,
+        dataset=default_dataset,
+        data_samples=default_dataset.train_data_sample_keys,
+        metadata={'foo': {"bar": "foo"}}
+    )
+    with pytest.raises(ValueError):
+        method_add(spec)
+
+    too_long = 'foo' * 40
+    # raise an error if the key is too long (more than 50 chars)
+    spec = method_create(
+        algo=algo,
+        dataset=default_dataset,
+        data_samples=default_dataset.train_data_sample_keys,
+        metadata={too_long: "bar"}
+    )
+    with pytest.raises(substra.exceptions.InvalidRequest):
+        method_add(spec)
+
+    # raise an error if the value is too long (more than 100 chars)
+    spec = method_create(
+        algo=algo,
+        dataset=default_dataset,
+        data_samples=default_dataset.train_data_sample_keys,
+        metadata={"foo": too_long}
+    )
+    with pytest.raises(substra.exceptions.InvalidRequest):
+        method_add(spec)
+
+
+def test_metadata_aggregatetuple(factory, client, default_dataset):
+    composite_algo_spec = factory.create_composite_algo()
+    composite_algo = client.add_composite_algo(composite_algo_spec)
+    composite_traintuple_spec = factory.create_composite_traintuple(
+        algo=composite_algo,
+        dataset=default_dataset,
+        data_samples=default_dataset.train_data_sample_keys,
+    )
+    composite_traintuple = client.add_composite_traintuple(composite_traintuple_spec)
```

Comment:
You can create an aggregatetuple without inmodels (see the discussion at https://github.com/SubstraFoundation/substra-backend/pull/189). This would simplify this test a great deal.